frontend/src/concepts/pipelines/content/pipelinesDetails/pipeline/SelectedTaskDrawerContent.tsx

```diff
@@ -28,7 +28,8 @@ const SelectedTaskDrawerContent: React.FC<SelectedTaskDrawerContentProps> = ({ t
     >
       <DrawerHead>
         <Title headingLevel="h2" size="xl">
-          {task.name}
+          {task.taskSpec?.metadata?.annotations['pipelines.kubeflow.org/task_display_name'] ||
```

Comment:
checking `PipelineRunTask` taskSpec is not optional (nor metadata inside `PipelineRunTaskSpec`) so either marked them as optional if you are sure they are not mandatory or remove the optional chaining. We should be mindful of our types.